globus_automate_client/flows_client.py

```diff
@@ -460,25 +463,35 @@ def list_flows(
 
         """
         self.authorizer = self.flow_management_authorizer
+
         params = {}
-        if roles is not None and len(roles) > 0:
-            params["filter_roles"] = ",".join(roles)
-        if role is not None:
+
+        # *role* takes precedence over *roles* (plural).
+        if role:
```

Comment:
The pattern for checking truthiness on one option followed by the next etc is common in this. As a rule, I am very wary of python truthiness checks. As these are parameters passed in so could have any value depending on what's coming from the caller, I'm even a bit more wary.

It _may_ be that truthiness is exactly the right check here (and elsewhere with this pattern), but I just wanted to double check this is being used with full thought to the possible values that could be passed and if, in these cases, truthiness accurately captures the intent.

Reply:
In the specific places where I used it, truthiness appears to be correct as the goal is to eliminate empty values like empty strings and empty lists.

It's true that the user could pass in arbitrary types and values in violation of the type annotations, but the original `if x is not None` checks wouldn't have caught or prevented that, either. For example, the user could pass in `slice(2, 7)` and potentially cause a TypeError during JSON encoding.

Please let me know if that's satisfactory reasoning, or if more discussion is required. I might be missing something you're trying to gently lead me to, as well! :grin:

Reply:
yep, that's fine. Just something I always watch out for, and if your thinking is that the various truthiness cases are what we want, I'm good. And, I agree, the `None` check had its own set of possible gotchas.